### DISCUSSION TITLE
babelmark 3

### DISCUSSION THREAD

**Discussion by kellyjosephprice** [micromark/micromark#81] [General] [answered]
Any chance of getting this on [babelmark 3](https://babelmark.github.io/)? I realize that may be a lot to ask as it appears it would require maintenance of a service.

**Comment by wooorm** [comment, marked as answer]
Im not sure how that would work! Does that require an "on-call" maintainer? 🤔

**Reply by ChristianMurphy** [reply]
https://github.com/babelmark/babelmark-registry gives some limited description of what setup is needed.
It looks like it needs to be hosted as a web service, no mention of an "on-call" maintainer though.
It may be worth opening an issue there for further clarification.

**Reply by kellyjosephprice** [reply]
> Im not sure how that would work! Does that require an "on-call" maintainer?

Maybe not 'on-call'. But it looks like babelmark expects to be able to make a `GET` with the markdown, and receive the HTML. If that service went down occasionally, oh well?

**Reply by wooorm** [reply]
I'm not sure I'd want to run and pay for such a server though 🤔 hmm

**Reply by wooorm** [reply]
there are also `js:` prefixes, maybe that loads some npm library? https://github.com/babelmark/babelmark-registry/blob/d950170fcc94ae9a999aaead284ecc3ebf855a41/registry.json#L185

**Reply by wooorm** [reply]
Looks like they’re hardcoded to get a client side library and use that: https://github.com/babelmark/babelmark-proxy/blame/ff39a0c7ab0e83a5c250b8dfbece6b3ffa3ee497/src/BabelMark/ApiController.cs#L69.

@kellyjosephprice, or @ChristianMurphy Could you write c# (I think it is)? and use an unpkg or snowpack or esm.sh bundle to use micromark?

**Reply by kellyjosephprice** [reply]
I've never written any C#. :grimacing: That might take me a bit to get `dotnet` setup.

**Reply by ChristianMurphy** [reply]
I don't have dotnet setup at the moment.
It also looks like it might take some tinkering, currently it runs JS on ChakraCore https://github.com/babelmark/babelmark-proxy/blob/ff39a0c7ab0e83a5c250b8dfbece6b3ffa3ee497/src/BabelMark/BabelMark.csproj#L14-L17
Not sure how good the ESM support is there, Node/V8 may need to be added https://github.com/Taritsyn/JavaScriptEngineSwitcher#js-engines
